android/src/main/java/com/appcuesreactnative/AppcuesReactNativeModule.kt

```diff
@@ -56,8 +55,44 @@ class AppcuesReactNativeModule(reactContext: ReactApplicationContext) : ReactCon
                         this.activityStorageMaxAge = activityStorageMaxAge.toInt()
                     }
                 }
+
+                this.analyticsListener = object: AnalyticsListener {
+                    override fun trackedAnalytic(type: AnalyticType, value: String?, properties: Map<String, Any>?, isInternal: Boolean) {
+                        val params = Arguments.createMap().apply {
+                            putString("analytic", type.name)
+                            putString("value", value ?: "")
+                            putMap("properties", writableMapOf(properties ?: emptyMap<String, Any>()))
+                            putBoolean("isInternal", isInternal)
+                        }
+
+                        reactApplicationContext
+                            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+                            .emit("analytics", params)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun writableMapOf(values: Map<*, *>): WritableMap {
+        val map = Arguments.createMap()
+        for ((anyKey, value) in values) {
+            val key = anyKey as? String ?: continue
+            when (value) {
+                null -> map.putNull(key)
+                is Boolean -> map.putBoolean(key, value)
+                is Double -> map.putDouble(key, value)
+                // The Android SDK passes dates as a Long Unix timestamp
+                is Long -> map.putDouble(key, value.toDouble())
+                is Int -> map.putInt(key, value)
+                is String -> map.putString(key, value)
+                is WritableMap -> map.putMap(key, value)
+                is WritableArray -> map.putArray(key, value)
+                is Map<*, *> -> map.putMap(key, writableMapOf(value))
```

Comment:
We don't allow Lists/Arrays passed from apps in the properties and we don't currently use them (survey stuff aside), so I didn't add support here. The survey data is a good question since at least for iOS, it'd gets the internal `StepState` object and wouldn't be able to do anything with that. The same problem would exist in the native SDKs. Maybe we need to JSON-ize the survey data before passing it into the native observer implementation?

Reply:
oh, ew, yeah forgot.  On Android the prop type is actually the `ExperienceStepFormItemState`, similar concept to iOS.  I guess that would work to conditionally attempt to serialize unknown types to JSON and deserialize back?  Does this circle back to just doing it for all the props at that point rather than having custom mapping? I guess does not have to.

Reply:
wait what if the array on iOS has a Date in it? does it need to go through the same reformatting type of logic that dictionaries do? down on line 146 of AppcuesReactNative.swift

Reply:
Good catch! Yes, it would